### PR TITLE
Fix crashing differ on large paragraphs

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (engine): No longer crash editor when trying to style a very long paragraph. Closes https://github.com/ckeditor/ckeditor5/issues/16819

---

### Additional information

It looks like 200k arguments, passed to `.push` method of the Array, causes Google Chrome crashing.  I switched it to plain for loop and it works fine.

### CODE REVIEW

**Review by niegowski — COMMENTED** [review]
Maybe a helper like our `spliceArray` could be a better solution?

https://github.com/ckeditor/ckeditor5/blob/08270aafcc6e7e4fb4f4a4d2ebbc7451f031d303/packages/ckeditor5-utils/src/splicearray.ts#L12-L33

**Comment by Mati365** [comment]
@pszczesniak / @niegowski I'd recommend sticking with a for loop to maintain consistency with other parts of `_generateDiffInstructionsFromChanges`. I don’t believe there will be any noticeable performance gains from optimizing this specific region, as it’s highly likely that it’s already JIT compiled to machine code

**Review by niegowski — CHANGES_REQUESTED** [review]
Please add an automatic test so we do not have a regression in the future.

**Comment by Mati365** [comment]
@niegowski Added

File: packages/ckeditor5-engine/src/model/differ.ts [hunk 1]
```diff
@@ -1520,7 +1520,17 @@ function _generateDiffInstructionsFromChanges( oldChildrenLength: number, change
 			// We removed `howMany` old nodes, update `oldChildrenHandled`.
 			oldChildrenHandled += change.howMany;
 		} else {
-			diff.push( ...'a'.repeat( change.howMany ).split( '' ) );
+			// Total maximum amount of arguments that can be passed to `Array.prototype.push` may be limited so we need to
+			// add them manually one by one to avoid this limit. However loop might be a bit slower than `push` method on
+			// smaller changesets so we need to decide which method to use based on the size of the change.
+			// See: https://github.com/ckeditor/ckeditor5/issues/16819
+			if ( change.howMany > 1500 ) {
```
Comment (pszczesniak) [inline packages/ckeditor5-engine/src/model/differ.ts:1527]: Could you write (here and/or in the issue) why we choose 1500?

It's based on some tests or it's some kind of random value?

Comment (Mati365) [reply packages/ckeditor5-engine/src/model/differ.ts:1527]: Limitation of args stack depends on OS / Browser etc. I chose some kind of random value.

**Review by niegowski — APPROVED** [review]

**Pull request merged**